strider/trapi.py

```diff
@@ -15,14 +15,18 @@ def merge_messages(messages: list[Message]) -> Message:
     for message in messages:
         knodes |= message["knowledge_graph"]["nodes"]
         kedges |= message["knowledge_graph"]["edges"]
-        results.extend(message["results"])
+        for result in message["results"]:
+            # Use pydantic model so that we can deduplicate
+            result_pydantic = Result.parse_obj(result)
```

Comment:
This is handy, but also encroaches on a nice separation we have between the web server that handles pydantic objects, and the inner methods, which don't. The other risk(?) here is that the results actually don't validate against the Result model, but in a way that Strider is OK with. We are deliberately lax in enforcing TRAPI compliance because we want to get away with using slightly naughty KPs if we can.

Reply:
In that case, how do we want to compare result objects? Do they need to be exactly equal, including extra fields? Or do we only care about specific fields, and if so which ones?

Reply:
I think we want to ignore the binding properties except for `id` and make sure list order doesn't matter.
```
{
    key: {bound["id"] for bound in value}
    for key, value in result["node_bindings"].items()
}
```